haystack/components/routers/conditional_router.py

```diff
@@ -146,15 +157,22 @@ def to_dict(self) -> Dict[str, Any]:
             # output_type needs to be serialized to a string
             route["output_type"] = serialize_type(route["output_type"])
 
-        return default_to_dict(self, routes=self.routes)
+        # set all filter callables to None for serialization
+        custom_filters = {filter_name: None for filter_name in self.custom_filters.keys()}
```

Comment:
We should correctly serialize and deserialize callbacks with [these functions](https://github.com/shadeMe/haystack/blob/3e0c532e0e8295d2f258d143b75d12894a9eb38c/haystack/utils/callable_serialization.py#L12).

Reply:
Revised to match the pattern in OutputAdapter.